sql/core/src/main/scala/org/apache/spark/sql/Arrow.scala

```diff
@@ -195,13 +112,135 @@ object Arrow {
+  override protected val valueVector: NullableSmallIntVector
+  = new NullableSmallIntVector("ShortValue", allocator)
+  override protected val valueMutator: NullableSmallIntVector#Mutator = valueVector.getMutator
+
+  override protected def writeNull(): Unit = valueMutator.setNull(count)
+  override protected def writeData(data: Any): Unit
+  = valueMutator.setSafe(count, data.asInstanceOf[Short])
+}
+
+class IntegerColumnWriter(allocator: BaseAllocator) extends PrimitiveColumnWriter(allocator) {
+  override protected val valueVector: NullableIntVector
+  = new NullableIntVector("IntValue", allocator)
+  override protected val valueMutator: NullableIntVector#Mutator = valueVector.getMutator
+
+  override protected def writeNull(): Unit = valueMutator.setNull(count)
+  override protected def writeData(data: Any): Unit
+  = valueMutator.setSafe(count, data.asInstanceOf[Int])
+}
+
+class LongColumnWriter(allocator: BaseAllocator) extends PrimitiveColumnWriter(allocator) {
+  override protected val valueVector: NullableBigIntVector
+  = new NullableBigIntVector("LongValue", allocator)
+  override protected val valueMutator: NullableBigIntVector#Mutator = valueVector.getMutator
+
+  override protected def writeNull(): Unit = valueMutator.setNull(count)
+  override protected def writeData(data: Any): Unit
+  = valueMutator.setSafe(count, data.asInstanceOf[Long])
+}
+
+class FloatColumnWriter(allocator: BaseAllocator) extends PrimitiveColumnWriter(allocator) {
+  override protected val valueVector: NullableFloat4Vector
+  = new NullableFloat4Vector("FloatValue", allocator)
+  override protected val valueMutator: NullableFloat4Vector#Mutator
+  = valueVector.getMutator
+
+  override protected def writeNull(): Unit = valueMutator.setNull(count)
+  override protected def writeData(data: Any): Unit
+  = valueMutator.setSafe(count, data.asInstanceOf[Float])
+}
+
+class DoubleColumnWriter(allocator: BaseAllocator) extends PrimitiveColumnWriter(allocator) {
+  override protected val valueVector: NullableFloat8Vector
+  = new NullableFloat8Vector("DoubleValue", allocator)
+  override protected val valueMutator: NullableFloat8Vector#Mutator
+  = valueVector.getMutator
+
+  override protected def writeNull(): Unit = valueMutator.setNull(count)
+  override protected def writeData(data: Any): Unit
+  = valueMutator.setSafe(count, data.asInstanceOf[Double])
+}
+
+class ByteColumnWriter(allocator: BaseAllocator) extends PrimitiveColumnWriter(allocator) {
+  override protected val valueVector: NullableUInt1Vector
+  = new NullableUInt1Vector("ByteValue", allocator)
+  override protected val valueMutator: NullableUInt1Vector#Mutator = valueVector.getMutator
+
+  override protected def writeNull(): Unit = valueMutator.setNull(count)
+  override protected def writeData(data: Any): Unit
+  = valueMutator.setSafe(count, data.asInstanceOf[Byte])
+}
+
+class UTF8StringColumnWriter(allocator: BaseAllocator) extends PrimitiveColumnWriter(allocator) {
```

Comment:
It has already been pretty much consolidated. Each column writers have different valueVector type, valueMutator type and writeData. The only part that is same among them is writeNull(), however, arrow doesn't have a nullable mutator interface, therefore this needs to be this way